src/Component/CodeEditor/CodeEditor.tsx

```diff
@@ -164,18 +166,50 @@ class CodeEditor extends React.Component<CodeEditorProps, CodeEditorState> {
     return parserOptions;
   }
 
+  onDownloadButtonClick = () => {
+    const {
+      activeParser,
+      value
+    } = this.state;
+    const {
+      style
+    } = this.props;
+    if (style) {
+      let type = 'application/json;charset=utf-8';
+      if (activeParser && activeParser.title === 'SLD Style Parser') {
```

Comment:
The comparison looks a bit fragile to my eyes...

Reply:
Indeed. But as there is just a small amount of StyleParsers i think this is our best opportunity to determine which parser was choosen.
Furthermore the static `title` prop was introduced to identify the parser type.